src/mon/OSDMonitor.cc

```diff
@@ -11594,3 +11631,41 @@ void OSDMonitor::_pool_op_reply(MonOpRequestRef op,
 					 ret, epoch, get_last_committed(), blp);
   mon->send_reply(op, reply);
 }
+
+void OSDMonitor::process_crush_errors()
+{
+  // Have we finished creating pgs?
+  if (!(creating_pgs.queue.size() == 0 && creating_pgs.pgs.size() == 0))
+    return;
+
+  mapping.check_new_crush_errors();
+
+  if (mapping.has_crush_errors()) {
+    auto pool_crush_errors = mapping.get_per_pool_crush_errors();
+    std::for_each(
+        pool_crush_errors->begin(), pool_crush_errors->end(),
+        [&](std::pair<long unsigned int, struct crush_errors_t> pair) {
+          if (pair.second.new_errors) {
+            mon->clog->warn()
```

Comment:
If we’re changing the health state of the monitor, why do we need to clog the details like this?

I’m particularly thinking about it because I suspect we will most often see errors in the OSD tree arrangement that invalidate *all* the pool mappings...